dotCMS/src/main/java/com/dotmarketing/viewtools/LanguageWebAPI.java

```diff
@@ -168,20 +166,33 @@ public String get(String key, List args) {
 	public String get(String key, String languageId) {
 		String value = null;
 		try {
-			Language lang = LanguageUtil.getUserLanguage(langAPI.getLanguage(languageId), request.getLocale());
-			value = langAPI.getStringKey(lang, key);
+
+			//Find the requested language
+			Language requestedLanguage;
+			try {
+				requestedLanguage = langAPI.getLanguage(languageId);
+			} catch (Exception e) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)